### PR TITLE
[show][mlnx] replace shell=True, replace xml

### PR DESCRIPTION
**Pull request by maipbui**
Signed-off-by: maipbui <maibui@microsoft.com>

#### What I did
`subprocess()` - when using with `shell=True` is dangerous. Using subprocess function without a static string can lead to command injection.
#### How I did it
`subprocess()` - use `shell=False` instead, use list of strings Ref: [https://semgrep.dev/docs/cheat-sheets/python-command-injection/#mitigation](https://semgrep.dev/docs/cheat-sheets/python-command-injection/#mitigation)
#### How to verify it
Add UT
Manual test
```
admin@***:~$ show platform mlnx issu
ISSU is enabled
admin@***:~$ show platform mlnx sniffer
sdk sniffer is disabled
```

### CODE REVIEW

File: show/plugins/mlnx.py [hunk 1]
```diff
@@ -46,9 +46,9 @@ def run_command(command, display_cmd=False, ignore_error=False, print_to_console
     """Run bash command and print output to stdout
     """
     if display_cmd == True:
-        click.echo(click.style("Running command: ", fg='cyan') + click.style(command, fg='green'))
+        click.echo(click.style("Running command: ", fg='cyan') + click.style(' '.join(command), fg='green'))
```
Comment (qiluo-msft) [inline show/plugins/mlnx.py:49]: >join [](http://example.com/codeflow?start=81&length=4)

`shlex.join` ?

**Comment by qiluo-msft** [comment]
@stepanblyschak @dprital Could you help review?

**Review by qiluo-msft — DISMISSED** [review]

**Comment by maipbui** [comment]
@stepanblyschak @dprital could you help review?

**Review by qiluo-msft — APPROVED** [review]

**Comment by maipbui** [comment]
@keboliu could you help review?

**Comment by keboliu** [comment]
> @keboliu could you help review?

could you please also provide the test result of command "show platform mlnx sniffer"?

**Comment by maipbui** [comment]
> > @keboliu could you help review?
> 
> could you please also provide the test result of command "show platform mlnx sniffer"?

I added in the PR description
```
admin@***:~$ show platform mlnx issu
ISSU is enabled
admin@***:~$ show platform mlnx sniffer
sdk sniffer is disabled
```

**Comment by maipbui** [comment]
@keboliu could you check PR description and review PR?

**Review by keboliu — APPROVED** [review]

**Pull request merged**